sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -167,10 +192,45 @@ case class Or(left: Expression, right: Expression)
       }
     }
   }
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): Code = {
+    val eval1 = left.gen(ctx)
+    val eval2 = right.gen(ctx)
+    s"""
+      ${eval1.code}
+      boolean ${ev.nullTerm} = false;
```

Comment:
this can be simplified right?
```
boolean ${ev.nullTerm} = ${eval1.nullTerm} || ${eval1.nullTerm};
boolean ${ev. primitiveTerm} = ${eval1.primitiveTerm} || ${eval1.primitiveTerm};
```